pacehrh/R/pace_utils.R

```diff
@@ -1,3 +1,54 @@
+.populationRateCategories <- c("femaleFertility",
+                               "maleFertility",
+                               "femaleMortality",
+                               "maleMortality")
+
+#' Generate A Long-Format Data Frame Of Suite Mortality and Fertility Rates
+#'
+#' @param results Results structure (as returned by [RunExperiments()])
+#' @param rateCategory Rate type ({"femaleFertility" | "maleFertility" | "femaleMortality" | "maleMortality"})
+#'
+#' @return Data frame
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' }
+GetSuiteRates <- function(results, rateCategory = .populationRateCategories[1]) {
+  if (is.null(results)){
+    warning(paste0("Null results parameter. Returning NULL."))
+    return(NULL)
+  }
+
+  if (!(rateCategory %in% .populationRateCategories)){
+    warning(paste0("Invalid population rate category (", rateCategory, ")"))
+    return(NULL)
+  }
+
+  trials <- names(results)
+
+  l <- lapply(trials, function(trial) {
+    t <-
+      results[[trial]][["PopulationRates"]][[rateCategory]][["ratesMatrix"]]
+    t <- as.data.table(t, keep.rownames = "Label")
```

Comment:
The model_input.xlsx seems to have missing labels, we need to agree to add this field as required (currently no validation rules are set for this https://institutefordiseasemodeling.github.io/PACE-HRH/articles/input-validation.html#sheet-popvalues

Reply:
Zero rates are both valid (fertility rates above/below fertile range), and the blank label is a signal to not show them in reports.